### PR TITLE
fix: Add tests to check lazy matching is applied and that the string …

### LINKED ISSUE #38190: Regex Exercises
This exercise:
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching

Can also be solve with this regex: ->      `/<h1>/`

* Browser Name: Firefox
* Browser Version: 
* Operating System:  Unbuntu

### PR DESCRIPTION
**Pull request by AshNaz87**
…"h1" is not matched

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #38190

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md [hunk 1]
```diff
@@ -26,7 +26,10 @@ Fix the regex <code>/&lt;.*&gt;/</code> to return the HTML tag <code>&lt;h1&gt;<
 tests:
   - text: The <code>result</code> variable should be an array with <code>&lt;h1&gt;</code> in it
     testString: assert(result[0] == '<h1>');
-
+  - text:  The myRegex variable should use lazy matching
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]:
```suggestion
  - text: <code>myRegex</code> should use lazy matching
```

Comment (AshNaz87) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]: @RandellDawson Changes made as requested

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md [hunk 2]
```diff
@@ -26,7 +26,10 @@ Fix the regex <code>/&lt;.*&gt;/</code> to return the HTML tag <code>&lt;h1&gt;<
 tests:
   - text: The <code>result</code> variable should be an array with <code>&lt;h1&gt;</code> in it
     testString: assert(result[0] == '<h1>');
-
+  - text: <code>myRegex</code> should use lazy matching
+    testString: assert(myRegex.test(myRegex.source))
```
Comment (moT01) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]: This test passes with the default code and the code in the screenshot.
I think something like this is probably enough to make sure lazy loading was used...
```suggestion
    testString: assert(/\?/g.test(myRegex))
```

**Review by moT01 — CHANGES_REQUESTED** [review]
This doesn't seem to fix the problem...
Screen Shot 2020-02-11 at 8 29 04 AM
All three tests pass with that regex ^

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md [hunk 3]
```diff
@@ -26,7 +26,10 @@ Fix the regex <code>/&lt;.*&gt;/</code> to return the HTML tag <code>&lt;h1&gt;<
...
+  - text: The <code>result</code> variable should not match the string 'h1'
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]:
```suggestion
  - text: <code>myRegex</code> should not include the string 'h1'
```
The user should make the existing regex lazy, not just hard-code a match for h1.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md [hunk 4]
```diff
@@ -26,7 +26,10 @@ Fix the regex <code>/&lt;.*&gt;/</code> to return the HTML tag <code>&lt;h1&gt;<
...
+    testString: assert(result[0] !== 'h1')
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]:
```suggestion
    testString: assert(!myRegex.source.match('h1'))
```
If the first test passes, we already know that the result is not 'h1', so this test should check the source.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md (hunk 2 above)
Comment (AshNaz87) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]: @moT01 Thanks for that. Made changes

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md [hunk 5]
```diff
@@ -26,7 +26,10 @@ Fix the regex <code>/&lt;.*&gt;/</code> to return the HTML tag <code>&lt;h1&gt;<
 tests:
   - text: The <code>result</code> variable should be an array with <code>&lt;h1&gt;</code> in it
     testString: assert(result[0] == '<h1>');
-
+  - text: <code>myRegex</code> should use lazy matching
+    testString: assert(/\?/g.test(myRegex));
+  - text: The <code>result</code> variable should not match the string 'h1'
+    testString: assert(result[0] !== 'h1');
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]:
```suggestion
    testString: assert(!myRegex.source.match('h1'))
```
If the first test passes, we already know that the result is not 'h1', so this test should check the source.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
@AshNaz87 I think my review comment came in at the exact moment you were addressing Tom's comments and got lost, so I've submitted it again.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md (hunk 5 above)
Comment (AshNaz87) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/find-characters-with-lazy-matching.english.md]: @ojeytonwilliams Applied

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Thanks, @AshNaz87.  LGTM.

@moT01 assuming the tests pass I think this can be merged.

**Review by moT01 — APPROVED** [review]
LGTM @AshNaz87 🎉 Thank you for contributing to freeCodeCamp! 🎉

**Pull request merged**